apps/api/src/controllers/post.ts

```diff
@@ -1,125 +1,117 @@
+    try {
+        const posts = await Post.find().populate('category');
+        res.status(200).json(posts);
+    } catch (error) {
+        const { message } = error;
+        res.status(500).json({ message });
+    }
 };
 
-const createPost: RequestHandler = (req: Request, res: Response) => {
-    const { title, image, description, category } = req.body;
-    if (!title) {
-        return res.status(400).json({ message: 'The title is required.' });
-    }
-    if (!image) {
-        return res.status(400).json({ message: 'The image is required.' });
-    }
-    if (!description) {
-        return res.status(400).json({ message: 'The description is required.' });
-    }
-    if (!category) {
-        return res.status(400).json({ message: 'The category is required.' });
+const createPost: RequestHandler = async (req: Request, res: Response) => {
+    try {
+        const newPost = await Post.create(req.body);
+        res.status(201).json(newPost);
+    } catch (error) {
+        const { message } = error;
+        res.status(500).json({ message });
     }
-
-    const newPost: IPost = {
-        id: Date.now().toString(),
-        title,
-        image,
-        description,
-        category,
-        comments: []
-    };
-    posts.push(newPost);
-
-    res.status(201).json(newPost);
 };
 
-const getPostByCategory: RequestHandler = (req: Request, res: Response) => {
+const getPostByCategory: RequestHandler = async (req: Request, res: Response) => {
     const { categoryId } = req.params;
 
-    const postsByCategory: IPost[] = posts.filter(item => item.category === categoryId);
-    
-    postsByCategory.map(postItem => postItem.category = getArrayItemByField(categories, 'id', postItem.category));
-
-    res.status(200).json(postsByCategory);
+    try {
+        const posts = await Post.where('category').equals(categoryId).populate('category');
+        res.status(200).json(posts);
+    } catch (error) {
+        const { message } = error;
+        res.status(500).json({ message });
+    }
 };
 
-const createPostComment: RequestHandler = (req: Request, res: Response) => {
+const createPostComment: RequestHandler = async (req: Request, res: Response) => {
     const { postId } = req.params;
-    const { author, content } = req.body;
-    const postIndex: number = getArrayItemIndexByField(posts, 'id', postId);
-    const updatedPost: IPost = { ...posts[postIndex] };
-    let newComment: IPostComment;
-
-    if (author && content) {
-        newComment = {
-            id: Date.now().toString(),
-            author,
-            content
-        };
-        updatedPost.comments.push(newComment.id);
-        comments.push(newComment);
+    try {
+        const currentPost = await Post.findById(postId);
+        const postComment = await PostComment.create(req.body);
+        currentPost.comments.push(postComment._id);
+        const updatedPost = await Post.findByIdAndUpdate(postId, currentPost, { new: true });
```

Comment:
Since we have an instance of the post in `currentPost` we can replace this line with `await currentPost.save();` and this will save the comment reference into the post